src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -314,10 +314,19 @@ private void addXApiImports(SurfaceTransformerContext context) {
     typeTable.getImports().remove(EMPTY_PROTO_PKG);
   }
 
-  private void addXExampleImports(SurfaceTransformerContext context) {
+  @VisibleForTesting
+  void addXExampleImports(SurfaceTransformerContext context) {
+    // Examples are different from the API. In particular, we use short declaration
+    // and so we omit most type names. We only need
+    //   context, to initialize the client
+    //   the VKit generated library, that's what the sample is for
+    //   the input types of the methods, to initialize the requests
     ModelTypeTable typeTable = context.getTypeTable();
+    typeTable.getImports().clear();
```

Comment:
I think it would be clearer for the imports to be reset in generateExample, where the imports are actually used. Then this method would purely "add" (thus performing an action that matches its name).